### DISCUSSION TITLE
How to enable debugging?

### DISCUSSION THREAD

**Discussion by majunze2001** [NVIDIA/open-gpu-kernel-modules#518] [Q&A]
Can I build the driver with debug symbols? I tried to gdb on the kernel, but it seemed the module did not export any symbols.

<img width="898" alt="image" src="https://github.com/NVIDIA/open-gpu-kernel-modules/assets/93145857/a47bcd97-236b-4d6d-a802-cbfa40a5749f">

**Comment by 0x401000Nu1l** [comment]
how do you use gdb to debug this driver? can you please give me some blog or doc to learn?

**Comment by ttabi** [comment]
You can debug it the same way you debug any Linux kernel driver for which you have the source code.  Note that you cannot debug the GSP-RM firmware itself, because Nvidia does not provide the tools for that.

**Comment by IgorTorrente-col** [comment]
`NV_BUILD_TYPE=debug make modules -j$(nproc)`

**Comment by kenorb** [comment]
Try:

    make modules -j$(nproc) DEBUG=1 NV_VERBOSE=1 CONFIG_DEBUG_INFO=y

**Comment by alokprasad** [comment]
How to see more debug prints in dmesg?
